src/routes/transactions/converters/summary.rs

```diff
@@ -142,11 +128,11 @@ impl ModuleTransaction {
 impl CreationTransaction {
     pub async fn to_transaction_summary(
         &self,
-        safe_address: &String,
+        safe_address: &str,
         info_provider: &(impl InfoProvider + Sync),
     ) -> TransactionSummary {
         TransactionSummary {
-            id: create_id!(super::super::models::ID_PREFIX_CREATION_TX, safe_address),
+            id: self.generate_id(safe_address),
```

Comment:
Why is the id for the creation transaction generated based on the `safe_address`?